### PR TITLE
Fb1063

### PR DESCRIPTION
**Pull request by nicola-spire**

### CODE REVIEW

File: applications/sudoku/source/index.tsx [hunk 1]
```diff
@@ -21,15 +22,17 @@ class Application extends React.Component<{}> {
...
                                              this.landingPage = e;
-                                           }
+                                          }
                                         }}/>;
             }}/>
           <Router.Route path='/game'
             render={() => {
               if(!this.gameModel) {
+                const board = sudoku.generateBoard();
+                board.set(0, 0, 0);
```
Comment (kranar) [inline applications/sudoku/source/index.tsx]: What is the purpose of setting 0,0?

File: library/source/pages/game_page/game_page.tsx [hunk 2]
```diff
@@ -226,13 +226,19 @@ export class GamePage extends React.Component<Properties, State> {
...
+                    onScoreUpdate={(score: number) => {
+                      this.props.model.score = score;
+                      this.forceUpdate();
+                    }}/>
       </VBoxLayout>);
   }
   private onGameFinish() {
-    this.props.model.save();
+    this.setState({redirect: this.props.exitUrl});
+
```
Comment (kranar) [inline library/source/pages/game_page/game_page.tsx]: Remove extra line.

File: library/source/pages/game_page/local_game_model.ts [hunk 3]
```diff
@@ -29,7 +30,12 @@ export class LocalGameModel extends GameModel {
   public get board(): Board {
     return this._board;
   }
-
+  public get score(): number {
+    return this._score;
+  }
+  public set score(score: number) {
```
Comment (kranar) [inline library/source/pages/game_page/local_game_model.ts]: Place blank lines between function declarations.

File: library/source/pages/result_page/result_page.tsx [hunk 4]
```diff
@@ -2,16 +2,15 @@ import {css, StyleSheet} from 'aphrodite';
 import * as History from 'history';
 import * as React from 'react';
 import * as Router from 'react-router-dom';
-import {Center, GameResult, HBoxLayout, Padding, VBoxLayout} from '../..';
+import {Center, GameModel, GameResult, HBoxLayout, Padding,
+        VBoxLayout} from '../..';
+import {LoadingPage} from '../';
```
Comment (kranar) [inline library/source/pages/result_page/result_page.tsx]: Remove the /. '..'

File: library/source/pages/result_page/result_page.tsx [hunk 5]
```diff
@@ -44,20 +44,30 @@ export class ResultPage extends React.Component<Properties, State> {
   constructor(props: Properties) {
     super(props);
     this.state = {
+      isLoading: true,
       redirect: null,
-      breakPoint: ResultPage.getBreakPoint()
+      breakPoint: ResultPage.getBreakPoint()
```
Comment (kranar) [inline library/source/pages/result_page/result_page.tsx]: Remove the extra space.

File: library/source/pages/result_page/result_page.tsx [hunk 6]
```diff
@@ -44,20 +44,30 @@ export class ResultPage extends React.Component<Properties, State> {
...
+        this.setState({isLoading: false});
+      });
   }
 
   public render(): JSX.Element {
     if(this.state.redirect) {
       return <Router.Redirect push to={this.state.redirect}/>;
     }
+    if(this.state.isLoading) {
+      return <LoadingPage/>
```
Comment (kranar) [inline library/source/pages/result_page/result_page.tsx]: Missing semi-colon.

File: library/source/pages/result_page/result_page.tsx [hunk 7]
```diff
@@ -212,6 +222,7 @@ export class ResultPage extends React.Component<Properties, State> {
       }
     }
   });
+  private gameResult: GameResult;
```
Comment (kranar) [inline library/source/pages/result_page/result_page.tsx]: Don't mix function declarations with member variables.

The order is public, protected, private, and then within each access specifier the order is functions, static variables, local variables.

File: tests/game_page_router_tester/source/index.tsx [hunk 8]
```diff
@@ -47,6 +49,7 @@ class Application extends React.Component<{}> {
   private landingPage: sudoku.LandingPage;
   private gameModel: sudoku.GameModel;
   private standingsModel: sudoku.StandingsModel;
+  private gameResult: sudoku.GameResult;
```
Comment (kranar) [inline tests/game_page_router_tester/source/index.tsx]: This is never used, remove it.

**Comment by nicola-spire** [comment]
I set 0, 0, on the game_page_router_tester so that I only need to fill one
cell to get to the results page. I fixed on the
applications/sudoku/source/index.tsx so that a new board is generated on a
new game click.

**Pull request merged**